### PR TITLE
NZ parser: Update oil for name change

### PR DESCRIPTION
**Pull request by drc38**
## Description
Updates the parser to match the latest descriptions from Transpower for the Oil category

### Double check

- [ ] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [ ] I have run `pnpx prettier --write .` and `poetry run format` to format my changes.

### CODE REVIEW

File: parsers/NZ.py [hunk 1]
```diff
@@ -95,7 +95,7 @@ def fetch_production(
         "datetime": datetime,
         "production": {
             "coal": productions.get("Coal", {"generation": None})["generation"],
-            "oil": productions.get("Liquid", {"generation": None})["generation"],
+            "oil": productions.get("Diesel\/Oil", {"generation": None})["generation"],
```
Comment (VIKTORVAV99) [inline parsers/NZ.py:98]:
```suggestion
            "oil": productions.get("Diesel/Oil", {"generation": None})["generation"],
```

File: parsers/NZ.py [hunk 2]
```diff
@@ -108,7 +108,7 @@ def fetch_production(
         },
         "capacity": {
             "coal": productions.get("Coal", {"capacity": None})["capacity"],
-            "oil": productions.get("Liquid", {"capacity": None})["capacity"],
+            "oil": productions.get("Diesel\/Oil", {"capacity": None})["capacity"],
```
Comment (VIKTORVAV99) [inline parsers/NZ.py:111]:
```suggestion
            "oil": productions.get("Diesel/Oil", {"capacity": None})["capacity"],
```

**Review by VIKTORVAV99 — CHANGES_REQUESTED** [review]
One tiny change needed then this will look good, thanks for doig this.

File: parsers/NZ.py (hunk 1 above)
Comment (drc38) [reply parsers/NZ.py]: Thanks, now fixed

**Review by VIKTORVAV99 — APPROVED** [review]
Thanks for updating this parser!

**Pull request merged**